add `OpInfo` for `torch.nn.pixel_shuffle`

pmeier opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #65681
* #65680
* #65469
* #65468
* __->__ #65467
* #65460

Differential Revision: [D31111697](https://our.internmc.facebook.com/intern/diff/D31111697)

cc @mruberry @jbschlosser @walterddr

pmeier committed:
add `OpInfo` for `torch.nn.pixel_shuffle`

[ghstack-poisoned]

zou3519 commented:
@zou3519 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31111697).

pmeier committed:
Update on "add `OpInfo` for `torch.nn.pixel_shuffle`"


Differential Revision: [D31111697](https://our.internmc.facebook.com/intern/diff/D31111697)

[ghstack-poisoned]

pmeier committed:
Update on "add `OpInfo` for `torch.nn.pixel_shuffle`"


Differential Revision: [D31111697](https://our.internmc.facebook.com/intern/diff/D31111697)

[ghstack-poisoned]

zou3519 commented:
@zou3519 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31111697).

pmeier committed:
Update on "add `OpInfo` for `torch.nn.pixel_shuffle`"


Differential Revision: [D31111697](https://our.internmc.facebook.com/intern/diff/D31111697)

[ghstack-poisoned]

zou3519 commented on torch/testing/_internal/common_methods_invocations.py:
```diff
@@ -5654,6 +5654,16 @@ def sample_inputs_pairwise_distance(op_info, device, dtype, requires_grad, **kwa
 
+def sample_inputs_pixel_shuffle(op_info, device, dtype, requires_grad, **kwargs):
+    return [
+        SampleInput(
+            make_tensor((1, 9, 2, 2), device=device, dtype=dtype, requires_grad=requires_grad),
+            kwargs=dict(upscale_factor=upscale_factor),
+        )
+        for upscale_factor in (1, 3)
```
The docs for pixel_shuffle https://pytorch.org/docs/stable/generated/torch.nn.PixelShuffle.html suggest that it accepts tensors that aren't 4D. Could we test something of size (9, 2, 2) and (2, 3, 9, 2, 2) ?

zou3519 reviewed (commented):
This looks pretty good. I had one suggestion to add some more shapes

pmeier committed:
Update on "add `OpInfo` for `torch.nn.pixel_shuffle`"


Differential Revision: [D31111697](https://our.internmc.facebook.com/intern/diff/D31111697)

cc albanD mruberry jbschlosser walterddr

[ghstack-poisoned]

pmeier committed:
Update on "add `OpInfo` for `torch.nn.pixel_shuffle`"


Differential Revision: [D31111697](https://our.internmc.facebook.com/intern/diff/D31111697)

cc albanD mruberry jbschlosser walterddr

[ghstack-poisoned]

pmeier committed:
Update on "add `OpInfo` for `torch.nn.pixel_shuffle`"


Differential Revision: [D31111697](https://our.internmc.facebook.com/intern/diff/D31111697)

cc albanD mruberry jbschlosser walterddr

[ghstack-poisoned]
